tests/system_tests_streaming_deliveries.py

```diff
@@ -0,0 +1,356 @@
+        result = self.ready_query
+        self.ready_query = False
+        return result
+
+    def is_ready_to_stream(self):
+        return self.ready_stream
+
+    def find_stats(self, response):
+        for record in response.results:
+            if record.name == 'M' + self.addr:
+                return (record.subscriberCount, record.remoteCount)
+        return (0,0)
+
+    def query_stats(self):
+        if self.proxy:
+            msg = self.proxy.query_addresses()
+            self.query_sender.send(msg)
+            #print("  SENT QUERY")
+
+    def start_stream(self):
+        msg = Message(body=None)
+        self.delivery = self.stream_sender.delivery(self.stream_sender.delivery_tag())
+        encoded = msg.encode()
+        self.stream_sender.stream(encoded)
+
+    def continue_stream(self, eos=False):
+        fragment = "Stream fragment %d" % self.fragment_id
+        self.fragment_id += 1
+        self.stream_sender.stream(bytes(fragment, 'ascii'))
+        if eos:
+            self.stream_sender.advance()
+
+    def on_link_opened(self, event):
+        if event.receiver == self.reply_receiver:
+            self.reply_addr  = event.receiver.remote_source.address
+            self.proxy       = MgmtMsgProxy(self.reply_addr)
+            self.ready_query = True
+
+    def on_message(self, event):
+        if event.receiver == self.reply_receiver:
+            response = self.proxy.response(event.message)
+            local, remote = self.find_stats(response)
+            #print("  RECEIVED REPLY (%d, %d)" % (local, remote))
+            if local == self.wait_local and remote == self.wait_remote:
+                #print("    READY")
+                self.ready_stream = True
+
+
+class InterleavedStreamsTest(MessagingHandler):
+    def __init__(self, sender_hosts, receiver_host, addr, fragments=10):
+        super(InterleavedStreamsTest, self).__init__()
+        self.sender_hosts   = sender_hosts  # List of (host, wait_local, wait_remote)
+        self.receiver_host  = receiver_host
+        self.addr           = addr
+        self.fragment_count = fragments
+
+        self.stream_senders     = []
+        self.stream_receivers   = {}    # {receiver => fragment count}
+        self.receiver_conn      = None
+        self.receiver           = None
+        self.error              = None
+        self.poll_timer         = None
+        self.n_ready_to_query   = 0
+        self.n_ready_to_stream  = 0
+        self.n_sent_fragments   = 0
+        self.n_received_streams = 0
+
+    def timeout(self):
+        self.error = "Timeout Expired - n_ready_to_query=%d, n_ready_to_stream=%d, n_sent_fragments=%d, n_received_streams=%d" % \
+            (self.n_ready_to_query, self.n_ready_to_stream, self.n_sent_fragments, self.n_received_streams)
+        self.receiver_conn.close()
+        for ss in self.stream_senders:
+            ss.close()
+        if self.poll_timer:
+            self.poll_timer.cancel()
+
+    def fail(self, error):
```

Comment:
Can we call this function bail() instead of fail() ? fail() with called with an error=None means the test has succeeded which sounds kinda weird.